python/cudf/cudf/tests/test_options.py

```diff
@@ -0,0 +1,79 @@
+# Copyright (c) 2022, NVIDIA CORPORATION.
+
+from contextlib import redirect_stdout
+from io import StringIO
+
+import pytest
+
+import cudf
+
+
+@pytest.fixture(scope="module", autouse=True)
+def empty_option_environment():
+    old_option_enviorment = cudf.options._OPTIONS
+    cudf.options._OPTIONS = {}
+    yield
+    cudf.options._OPTIONS = old_option_enviorment
+
+
+@pytest.fixture(scope="module")
+def odd_option(empty_option_environment):
+    def validator(x):
+        if not x % 2 == 1:
+            raise ValueError(f"Invalid option {x}")
+
+    cudf.options._register_option(
+        "odd_option",
+        1,
+        "An odd option.",
+        validator,
+    )
+    yield
+    del cudf.options._OPTIONS["odd_option"]
+
+
+@pytest.fixture(scope="module")
+def even_option(empty_option_environment):
+    def validator(x):
+        if not x % 2 == 0:
+            raise ValueError(f"Invalid option {x}")
+
+    cudf.options._register_option(
+        "even_option", 0, "An even option.", validator
+    )
+    yield
+    del cudf.options._OPTIONS["even_option"]
+
+
+def test_option_get_set(odd_option):
+    assert cudf.get_option("odd_option") == 1
+    cudf.set_option("odd_option", 101)
+    assert cudf.get_option("odd_option") == 101
+    # Restore default value for other tests
+    cudf.set_option("odd_option", 1)
```

Comment:
This need to reset the data indicates to me that we shouldn't be using a module-level fixture for `odd_option`. This should be more tightly scoped to test function level.

Reply:
That makes sense. In this way every pytest worker will reset the option for each test case.